crates/cli/src/codegen/mod.rs

```diff
@@ -370,11 +401,12 @@ impl Generator {
             CodeGenType::Static => {
                 // Remove no longer necessary exports.
                 module.exports.remove("canonical_abi_realloc")?;
-                // User plugins won't have an `eval_bytecode` function that
-                // Javy "owns".
-                if !self.plugin.is_user_plugin() {
+
+                // Only internal plugins expose eval_bytecode function.
+                if matches!(self.plugin.kind(), PluginKind::Internal(..)) {
```

Comment:
Similarly here, I was hoping this would look more like:

```rust
if self.linking_v2_plugin || self.linking_default_plugin {
```

with a similar setter for `linking_default_plugin` that we could gate behind a feature.